### PR TITLE
(docs) Use Presidio across Anthropic, Bedrock, VertexAI, Azure OpenAI, etc. w/ LiteLLM Proxy

### PR DESCRIPTION
**Pull request by krrishdholakia**
## Change Description

Adds a sample doc for setting up Presidio (docker image) with LiteLLM Proxy for implementing PII Masking across 100+ LLMs. 

## Issue reference

n/a 

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] My PR contains documentation updates / additions if required

cc: @omri374 @SharonHart

### CODE REVIEW

**Comment by omri374** [comment]
Thanks! We'll review this soon

File: docs/samples/docker/litellm.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+
+```
+litellm --config /path/to/config.yaml
+```
+
+
+This will mask the input going to the llm provider
+
+<Image img={require('../../img/presidio_screenshot.png')} />
```
Comment (SharonHart) [inline docs/samples/docker/litellm.md:32]: not referenced well? I assume its taken from your repo?

File: docs/samples/docker/litellm.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
+# LiteLLM (OpenAI Proxy) with Presidio
+
+LiteLLM supports [Microsoft Presidio](https://github.com/microsoft/presidio/) for PII masking. 
+
+[👉 **Refer to LiteLLM Docs for detailed guide**](https://docs.litellm.ai/docs/proxy/pii_masking)
```
Comment (SharonHart) [inline docs/samples/docker/litellm.md:5]: I would make it a bit more explicit that the below instructions are based a basic quick start running (f.e. LiteLLM running as docker on port 4000 is not mentioned here or on the pii masking quickstart)

File: mkdocs.yml [hunk 3]
```diff
@@ -53,6 +53,7 @@ nav:
   - Samples: 
     - Home: samples/index.md  
     - Selected samples:
+      - LiteLLM (OpenAI Proxy): samples/docker/litellm.md
```
Comment (SharonHart) [inline mkdocs.yml:56]: I see it as less common from the other samples, would move it a bit down 😃

**Review by SharonHart — COMMENTED** [review]
Looks great, left a few nits
Thanks for the contribution!

**Comment by krrishdholakia** [comment]
@SharonHart @omri374 updated based on feedback - any remaining blockers for merge?

File: mkdocs.yml [hunk 4]
```diff
@@ -57,6 +57,7 @@ nav:
       - NER model configuration: samples/python/ner_model_configuration.ipynb
       - Presidio Structured Basic Usage: samples/python/example_structured.ipynb
       - Using an allow list with image redaction: samples/python/image_redaction_allow_list_approach.ipynb
+      - LiteLLM (OpenAI Proxy): samples/docker/litellm.md
```
Comment (omri374) [inline mkdocs.yml:60]: Please add to the index.md on the samples folder. These are only the most fundamental notebooks for Presidio.

File: docs/samples/docker/litellm.md [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+
+3. LLM Response: "Hey [PERSON], nice to meet you!"
+
+4. User Response: "Hey Jane Doe, nice to meet you!"
+
+## Ad-hoc recognizers 
+
+Send ad-hoc recognizers to presidio `/analyze` by passing a json file to the proxy 
+
+[**Example** ad-hoc recognizer](../../../../litellm/proxy/hooks/example_presidio_ad_hoc_recognizer.json)
```
Comment (omri374) [inline docs/samples/docker/litellm.md:71]: Broken link

File: mkdocs.yml (hunk 4 above)
Comment (krrishdholakia) [reply mkdocs.yml]: resolved

File: docs/samples/docker/litellm.md (hunk 5 above)
Comment (krrishdholakia) [reply docs/samples/docker/litellm.md]: fixed

**Review by SharonHart — DISMISSED** [review]

File: docs/samples/docker/litellm.md [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+
+And search for any log starting with `Presidio PII Masking`, example:
+```bash
+Presidio PII Masking: Redacted pii message: <PERSON> AHV number is <AHV_NUMBER>. Zip code: <US_DRIVER_LICENSE>
+```
+
+
+## Turn on/off per key 
+
+Turn off PII masking for a given key.
```
Comment (omri374) [inline docs/samples/docker/litellm.md:102]: Can you add more information about what keys are and what turning them on / off does?

File: docs/samples/docker/litellm.md [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+--header 'Content-Type: application/json' \
+--data '{
+    "permissions": {"pii": false}
+}'
+```
+
+
+## Turn on/off per request 
+
+The proxy support 2 request-level PII controls:
```
Comment (omri374) [inline docs/samples/docker/litellm.md:118]:
```suggestion
The proxy supports 2 request-level PII controls:
```

File: docs/samples/docker/litellm.md [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+
+- *no-pii*: Optional(bool) - Allow user to turn off pii masking per request.
+- *output_parse_pii*: Optional(bool) - Allow user to turn off pii output parsing per request.
```
Comment (omri374) [inline docs/samples/docker/litellm.md:121]: What is the difference between pii masking and pii output parsing?

**Review by omri374 — COMMENTED** [review]
Thanks! Great addition.

I would suggest to add an explanation on the top specifying the user cases for using presidio with litellm, and how this flow works in high level.

**Comment by krrishdholakia** [comment]
Hey @omri374 @SharonHart  updated based on feedback - is this okay to merge?

**Review by omri374 — APPROVED** [review]
Thanks! Great contribution and addition

**Comment by krrishdholakia** [comment]
Hey @omri374 @SharonHart i still don't see litellm on the docs - https://microsoft.github.io/presidio/samples/

Did i put this somewhere wrong?

**Comment by omri374** [comment]
Hey @krrishdholakia, the website is now updated and contains the litellm docs. Thanks for the note!

**Pull request merged**